### ISSUE TITLE
Implement method to get full activity object

### ISSUE THREAD

**Issue by garemoko** [RusticiSoftware/TinCanPHP#24]
Something along these lines should work:

```
    public function retrieveFullActivityObject($activityid) {
        $response = $this->sendRequest(
            'GET',
            'activities',
            array(
                'params' => array(
                    'activityId' => $activityid,
                ),
                'ignore404' => true,
                'headers' => array(
                    'Accept-language: ' . $_SERVER['HTTP_ACCEPT_LANGUAGE'] . ', *'
                )
            )
        );
        if ($response->success) {
            $response->content = new Activity(json_decode($response->content, true));
        }
        return $response;
    }
```

**Comment by dacdave** [comment]
I get that this naming comes from the section naming in the spec, "Full Activity Object GET." However, there is a parallel operation defined a few sections later on the "agents" endpoint where the title for that section is "Combined Information GET."  I don't think that we want a method named retrieveCombinedInformation($agentid),

I propose that to use the name retrieveActivity (or possibly retrieveFullActivity, but not retrieveFullActivityObject) for the "activities" endpoint and retrievePerson (or possibly retrieveCombinedPerson) for the "agents" endpoint. I suggest retrievePerson over retrieveAgent to acknowledge the text in the spec that says, "Return a special, Person Object for a specified Agent."

**Comment by garemoko** [comment]
Sounds like you've looked into this, considered it carefully and made sensible suggestions. 

I'm happy with that name. +1

**Comment by dacdave** [comment]
+1 for new names. Since the request to the "agents" endpoint returns this special Person object that may be a combination of multiple personas (did I say that right?), then I suggest the name retrieveCombinedAgent(Agent agent) for the "agents" endpoint. For symmetry, I suggest retrieveFullActivity(Activity activity) for the "activities" endpoint.

I will make these changes to my code and update my pull request.

**Comment by dacdave** [comment]
Correction,

I meant retrieveCombinedPerson(Agent agent) for the GET call to the "agents" endpoint.

**Comment by brianjmiller** [comment]
Been thinking on this a bit, and in light of @garemoko's PR with `retrieveActivity` I think we want to go that route. I think `retrievePerson` probably makes the most sense (after some internal, offline discussion). Using `FullActivity` and/or `CombinedPerson` implies that there may or should exist a lesser Activity or a non-combined Person fetch, and neither of those can exist so I don't think the added descriptor is helpful and may be the opposite. I'd like to maintain parity with the resources themselves, IOW `/agents` which makes sense for `retrieveAgents` but that falls apart on the other side because `retrieveActivities` *really* doesn't make sense since you are only getting one. The "Full" part of the method makes sense in the case when you are starting with an Activity object to begin with, but I can see us wanting to support a string call where the argument to the method isn't already an object instance, in that case we can fetch and then construct one and that method should be named the same. A Person is by its nature "Combined" so I don't think that is necessary.

**Comment by dacdave** [comment]
Is there a difference between retrieveAgent and retrieveAgents? Can we agree on retrievePerson, for the agents endpoint, where person may be a single agent identifier or an array of identifiers associated in some proprietary way that the LRS maps these to the same "person?"

For the activities endpoint, I vote for the simple solution of retrieveActivity (note the singular).

**Comment by garemoko** [comment]
I've used retrievePerson and retrieveActivity [here](https://github.com/RusticiSoftware/TinCanPHP/pull/28/).

**Comment by dacdave** [comment]
@brianjmiller, are we agreed on these names?

**Comment by brianjmiller** [comment]
@dacdave yep, we'll use `retrievePerson` and `retrieveActivity`.

Want to be careful with this wording: "where person may be a single agent identifier or an array of identifiers"... a person is an object with a set of properties whose value is an array of identifiers.

**Comment by brianjmiller** [comment]
Resolved as part of #41 I believe.
